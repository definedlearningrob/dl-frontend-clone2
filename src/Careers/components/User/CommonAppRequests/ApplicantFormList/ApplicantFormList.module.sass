@import '@shared/stylesheets/config.sass'

$grid-template-columns: 3fr 1fr 150px 1fr 64px
$grid-template-columns-xxxl: 3fr 1fr 1fr 0.75fr 72px
$custom-button-height: 32px

.table
  height: 100%

  :global(.shared-table__head),
  :global(.shared-table__body)
    grid-template-columns: $grid-template-columns

    +media-breakpoint-up(xxxl)
      grid-template-columns: $grid-template-columns-xxxl

.emptyTableWrapper
  flex: 0 1 auto

.tableHeadCell
  color: $color-font-primary
  font-weight: $font-weight-regular
  line-height: $line-height-lg
  padding: $m-xs $m-x
  font-size: $font-size-xs

  +media-breakpoint-up(xxxl)
    padding: $m-sm

.tableRow:hover
  :global(.tooltip-wrapper)
    visibility: visible !important

  td
    background-color: $color-primary-200

  :global(#radix-trigger)
    visibility: visible !important

.formName
  color: $color-primary-500
  font-weight: $font-weight-bold
  line-height: $line-height-base

.emptyFormsMessage
  font-size: $font-size-xs
  padding: $m-x
  text-align: center

  +media-breakpoint-up(xxxl)
    font-size: $font-size-sm
    padding: $m-sm
