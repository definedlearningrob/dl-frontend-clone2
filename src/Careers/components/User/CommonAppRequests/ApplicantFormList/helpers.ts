import { COMMON_APP_FORM_TYPES } from '@dc/screens/UserApp/CommonApp/CommonAppRequests/types';

export const UNSUBMITABLE_FORMS = [
  COMMON_APP_FORM_TYPES.COUNSELOR_FEE_WAIVER,
  COMMON_APP_FORM_TYPES.COUNSELOR_EARLY_DECISION,
  COMMON_APP_FORM_TYPES.COUNSELOR_EARLY_DECISION_2,
];

export const MEMBER_SPECIFIC_FORMS = [
  COMMON_APP_FORM_TYPES.COUNSELOR_FINAL_REPORT,
  COMMON_APP_FORM_TYPES.COUNSELOR_EARLY_DECISION,
  COMMON_APP_FORM_TYPES.COUNSELOR_EARLY_DECISION_2,
];
