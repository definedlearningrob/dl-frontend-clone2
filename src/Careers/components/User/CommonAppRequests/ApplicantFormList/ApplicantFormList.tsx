import cx from 'classnames';
import isEmpty from 'lodash-es/isEmpty';
import { ReactNode, useMemo } from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import dayjs from 'dayjs';

import SharedTable from '@dc/shared/Table/Table';
import { APPLICATION_FORM_TYPE, COMMON_APP_FORM_STATUS } from '@dc/resources/enums';
import { StatusBadge } from '@dc/components/User/CommonAppRequests/ApplicantFormList/StatusBadge';
import {
  COMMON_APP_FORM_TYPES,
  CommonAppForm,
  RECOMMENDER_TYPES,
} from '@dc/screens/UserApp/CommonApp/CommonAppRequests/types';

import { DateWithTooltip } from '@shared/components/DateWithTooltip/DateWithTooltip';
import { NeedsAttentionMarker } from '@shared/components/NeedsAttentionMarker/NeedsAttentionMarker';

import styles from './ApplicantFormList.module.sass';
import { useLockedFormStatus } from './useLockedFormStatus';
import { ApplicantFormContextMenu } from './ApplicantFormContextMenu';

type Column = {
  render: (form: CommonAppForm, index: number) => ReactNode;
  title: string;
};

type Props = {
  forms: CommonAppForm[];
};

export const ApplicantFormList = ({ forms }: Props) => {
  const { t } = useTranslation();
  const history = useHistory();
  const { pathname } = useLocation();
  const hasNoData = isEmpty(forms);
  const { getLockedFormStatus } = useLockedFormStatus(forms);

  const navigateToForm = (form: CommonAppForm) => {
    const isFormLocked = getIsFormLockedStatus(form);

    if (isFormLocked) return;

    switch (form.formType) {
      case COMMON_APP_FORM_TYPES.TEACHER_RECOMMENDATION:
        history.push(`${pathname}/forms/teacher-recommendation`);
        break;
      case COMMON_APP_FORM_TYPES.COUNSELOR_EARLY_DECISION:
        history.push(`${pathname}/forms/early-decision`);
        break;
      case COMMON_APP_FORM_TYPES.COUNSELOR_EARLY_DECISION_2:
        history.push(`${pathname}/forms/early-decision2`);
        break;
      case COMMON_APP_FORM_TYPES.COUNSELOR_FEE_WAIVER:
        history.push(`${pathname}/forms/fee-waiver`);
        break;
      case COMMON_APP_FORM_TYPES.COUNSELOR_FINAL_REPORT:
        history.push(`${pathname}/forms/final-report`);
        break;
      case COMMON_APP_FORM_TYPES.COUNSELOR_MIDYEAR_REPORT:
        history.push(`${pathname}/forms/mid-year-report`);
        break;
      case COMMON_APP_FORM_TYPES.COUNSELOR_RECOMMENDATION:
        history.push(`${pathname}/forms/counselor-recommendation`);
        break;
      case COMMON_APP_FORM_TYPES.COUNSELOR_OPTIONAL_REPORT:
        history.push(`${pathname}/forms/optional-report`);
        break;
      case COMMON_APP_FORM_TYPES.COUNSELOR_OPTIONAL_REPORT_2:
        history.push(`${pathname}/forms/optional-report2`);
        break;
      case COMMON_APP_FORM_TYPES.COUNSELOR_SECONDARY_REPORT:
        history.push(`${pathname}/forms/secondary-report`);
        break;
    }
  };

  const getIsFormLockedStatus = (form: CommonAppForm) => {
    const formLockedStatus = getLockedFormStatus(form.formType);

    return formLockedStatus.isLocked;
  };

  const columns: Column[] = useMemo(
    () => [
      {
        title: t('user.postSecondary.commonAppRequests.applicationFormList.formName'),
        render: ({ formType, deadline, status }: CommonAppForm) => {
          const monthBeforeDeadline = dayjs(deadline).subtract(1, 'month');
          const isSubmitted = status === COMMON_APP_FORM_STATUS.SUBMITTED;

          const needsAttention = !isSubmitted && dayjs().isAfter(monthBeforeDeadline);

          return (
            <div>
              {needsAttention && (
                <NeedsAttentionMarker text={t('user.postSecondary.commonAppRequests.deadline')} />
              )}
              <div className={styles.formName}>{APPLICATION_FORM_TYPE[formType]}</div>
            </div>
          );
        },
      },
      {
        title: t('user.postSecondary.commonAppRequests.applicationFormList.recommenderType'),
        render: (form: CommonAppForm) => (
          <div className={styles.recommenderType}>
            {t(
              `user.postSecondary.commonAppRequests.applicationFormList.formValues.${
                form.formType.includes(RECOMMENDER_TYPES.TEACHER) ? 'teacher' : 'counselor'
              }`
            )}
          </div>
        ),
      },
      {
        title: t('user.postSecondary.commonAppRequests.applicationFormList.status'),
        render: (form: CommonAppForm) => (
          <StatusBadge
            formType={form.formType}
            status={form.status}
            {...getLockedFormStatus(form.formType)}
          />
        ),
      },
      {
        title: t('user.postSecondary.commonAppRequests.deadline'),
        render: ({ deadline }: CommonAppForm) => <DateWithTooltip date={deadline} />,
      },
      {
        title: '',
        render: (form: CommonAppForm) => (
          <ApplicantFormContextMenu
            form={form}
            isFormLocked={getLockedFormStatus(form.formType).isLocked}
            navigateToForm={navigateToForm}
          />
        ),
      },
    ],
    [forms]
  );

  const tableColumnClassnames = cx(
    'flex items-center relative',
    '!px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md',
    'last:!pr-base !py-sm',
    'text-xs cursor-pointer'
  );

  return (
    <>
      <SharedTable
        tableClassname={styles.table}
        tableWrapperClassname={cx({
          [styles.emptyTableWrapper]: hasNoData,
        })}>
        <SharedTable.Head
          cols={columns}
          columnClassname={cx(styles.tableHeadCell, tableColumnClassnames)}
        />
        <SharedTable.Body
          cols={columns}
          columnClassname={tableColumnClassnames}
          data={forms}
          rowClassname={cx('group/row', styles.tableRow)}
          onRowClick={(_, form) => navigateToForm(form)}
        />
      </SharedTable>
      {hasNoData && (
        <div className={styles.emptyFormsMessage}>
          {t('user.postSecondary.commonAppRequests.applicationFormList.empty')}
        </div>
      )}
    </>
  );
};
