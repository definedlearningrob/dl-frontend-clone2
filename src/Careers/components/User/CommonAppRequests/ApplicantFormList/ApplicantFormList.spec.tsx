import userEvent from '@testing-library/user-event';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { merge, set } from 'lodash-es';
import { screen, within } from '@testing-library/dom';
import { Route } from 'react-router-dom';

import { TUserInfo } from '@dc/graphql/user/queries/userInfo';
import { UserInfoProvider } from '@dc/hooks/useUserInfo';
import {
  CommonAppForm,
  COMMON_APP_FORM_TYPES,
} from '@dc/screens/UserApp/CommonApp/CommonAppRequests/types';
import { renderWithRouterAndReduxProvider } from '@dc/utils/test';
import { userInfoMock } from '@dc/tests/mocks/userMocks';
import { COMMON_APP_FORM_STATUS, SYNC_STATUS } from '@dc/resources/enums';
import {
  StudentApplicationsData,
  STUDENT_APPLICATIONS_QUERY,
} from '@dc/graphql/user/queries/studentApplications';

import { Roles } from '@shared/resources/enums';

import { ApplicantFormList } from './ApplicantFormList';

const commonAppStudentApplicationResponse = {
  data: {
    studentApplications: [
      {
        forms: [
          {
            formType: COMMON_APP_FORM_TYPES.TEACHER_RECOMMENDATION,
            status: COMMON_APP_FORM_STATUS.NOT_STARTED,
          },
          {
            formType: COMMON_APP_FORM_TYPES.COUNSELOR_SECONDARY_REPORT,
            status: COMMON_APP_FORM_STATUS.NOT_STARTED,
          },
          {
            formType: COMMON_APP_FORM_TYPES.COUNSELOR_OPTIONAL_REPORT,
            status: COMMON_APP_FORM_STATUS.NOT_STARTED,
          },
          {
            formType: COMMON_APP_FORM_TYPES.COUNSELOR_OPTIONAL_REPORT_2,
            status: COMMON_APP_FORM_STATUS.NOT_STARTED,
          },
          {
            formType: COMMON_APP_FORM_TYPES.COUNSELOR_MIDYEAR_REPORT,
            status: COMMON_APP_FORM_STATUS.NOT_STARTED,
          },
          {
            formType: COMMON_APP_FORM_TYPES.COUNSELOR_FINAL_REPORT,
            status: COMMON_APP_FORM_STATUS.NOT_STARTED,
          },
        ],
        institution: {
          id: '2',
          name: 'University of Alabama at Birmingham',
        },
      },
    ],
  },
};
const commonAppStudentApplicationMock: MockedResponse<StudentApplicationsData> = {
  request: {
    query: STUDENT_APPLICATIONS_QUERY,
    variables: {
      studentUuid: '0123',
    },
  },
  result: commonAppStudentApplicationResponse,
};

const counselorForms = [
  {
    formType: COMMON_APP_FORM_TYPES.COUNSELOR_SECONDARY_REPORT,
    status: COMMON_APP_FORM_STATUS.NOT_STARTED,
    deadline: null,
  },
  {
    formType: COMMON_APP_FORM_TYPES.COUNSELOR_OPTIONAL_REPORT,
    status: COMMON_APP_FORM_STATUS.NOT_STARTED,
    deadline: null,
  },
  {
    formType: COMMON_APP_FORM_TYPES.COUNSELOR_OPTIONAL_REPORT_2,
    status: COMMON_APP_FORM_STATUS.NOT_STARTED,
    deadline: '2023-11-01T09:00:00Z',
  },
  {
    formType: COMMON_APP_FORM_TYPES.COUNSELOR_MIDYEAR_REPORT,
    status: COMMON_APP_FORM_STATUS.NOT_STARTED,
    deadline: '2023-11-02T12:00:00Z',
  },
  {
    formType: COMMON_APP_FORM_TYPES.COUNSELOR_FINAL_REPORT,
    status: COMMON_APP_FORM_STATUS.NOT_STARTED,
    deadline: null,
  },
  {
    formType: COMMON_APP_FORM_TYPES.COUNSELOR_RECOMMENDATION,
    status: COMMON_APP_FORM_STATUS.SUBMITTED,
    deadline: null,
  },
];
const teacherForms = [
  {
    formType: COMMON_APP_FORM_TYPES.TEACHER_RECOMMENDATION,
    status: COMMON_APP_FORM_STATUS.NOT_STARTED,
    deadline: '2023-11-12T04:00:00Z',
  },
];

const syncStatus = {
  status: SYNC_STATUS.COMPLETED,
  lastSyncedAt: '2021-08-09T12:00:00.000Z',
};

const getCounselorUserInfo = (hasProfileCompleted: boolean = false) => ({
  permissions: { counselor: true, wblAdmin: false, canImpersonate: false, canBrowseReports: false },
  role: Roles.TEACHER,
  commonAppData: {
    hasRecommenderInvitation: true,
    hasTeacherInvitation: false,
    hasCounselorInvitation: true,
    hasCounselorProfileFormCompleted: hasProfileCompleted,
    hasTeacherProfileFormCompleted: true,
    syncStatus,
  },
});
const teacherUserInfo = {
  permissions: { counselor: true, wblAdmin: false, canImpersonate: false, canBrowseReports: false },
  role: Roles.TEACHER,
  commonAppData: {
    hasRecommenderInvitation: true,
    hasTeacherInvitation: true,
    hasCounselorInvitation: false,
    hasCounselorProfileFormCompleted: true,
    hasTeacherProfileFormCompleted: false,
    syncStatus,
  },
};

type UserInfo = Partial<TUserInfo>;
type Params = {
  userInfo: UserInfo;
  forms: CommonAppForm[];
};

jest.mock('@radix-ui/react-dropdown-menu', () => ({
  Root: jest.fn((props) => <div {...props} />),
  Trigger: jest.fn((props) => <button {...props} />),
  Portal: jest.fn((props) => <div {...props} />),
  Content: jest.fn((props) => <div {...props} />),
  Item: jest.fn((props) => <div {...props} />),
}));

const renderComponent = ({ userInfo, forms }: Params) => {
  const modifiedUserInfo = merge(userInfoMock.result.data.userInfo, userInfo);

  const modifiedUserInfoMock = { ...userInfoMock };

  set(modifiedUserInfoMock, 'result.data.userInfo', modifiedUserInfo);

  return renderWithRouterAndReduxProvider(
    <MockedProvider mocks={[modifiedUserInfoMock, commonAppStudentApplicationMock]}>
      <UserInfoProvider value={{ userInfo: modifiedUserInfo }}>
        <Route path='/post-secondary/commonapp-requests/:studentUuid'>
          <ApplicantFormList forms={forms} />
        </Route>
      </UserInfoProvider>
    </MockedProvider>,
    {
      route: '/post-secondary/commonapp-requests/0123',
    }
  );
};

describe('ApplicantFormList', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', async () => {
    const { container } = renderComponent({
      userInfo: getCounselorUserInfo(),
      forms: counselorForms,
    });

    await screen.findByRole('table');

    expect(container).toMatchSnapshot();
  });

  it('displays locked status for teacher forms when teacher profile is not completed', async () => {
    const { history } = renderComponent({ userInfo: teacherUserInfo, forms: teacherForms });
    history.push = jest.fn();

    await screen.findByRole('table');

    const teacherRecommendationRow = screen.getByRole('row', {
      name: /Teacher recommendation/,
    });

    expect(
      within(teacherRecommendationRow).getByRole('cell', { name: 'Locked' })
    ).toBeInTheDocument();

    userEvent.click(teacherRecommendationRow);

    expect(history.push).toHaveBeenCalledTimes(0);
  });

  it('displays locked status for counselor forms when counselor profile is not completed', async () => {
    const { history } = renderComponent({
      userInfo: getCounselorUserInfo(),
      forms: counselorForms,
    });
    history.push = jest.fn();
    await screen.findByRole('table');

    const [, ...tableBodyRows] = screen.getAllByRole('row');

    tableBodyRows.forEach((tableRow) => {
      expect(within(tableRow).getByRole('cell', { name: 'Locked' })).toBeInTheDocument();
      userEvent.click(tableRow);
    });

    expect(history.push).toHaveBeenCalledTimes(0);
  });

  it.each([
    ['Not started', 'Counselor secondary report'],
    ['Locked', 'Counselor optional report'],
    ['Locked', 'Counselor optional report 2'],
    ['Locked', 'Counselor midyear report'],
    ['Locked', 'Counselor final report'],
    ['Submitted', 'Counselor recommendation'],
  ])(
    'displays "%s" status for "%s" when counselor secondary report is not completed',
    async (status, formName) => {
      const { history } = renderComponent({
        userInfo: getCounselorUserInfo(true),
        forms: counselorForms,
      });
      history.push = jest.fn();
      await screen.findByRole('table');

      const [reportRow] = screen.getAllByRole('row', {
        name: new RegExp(formName),
      });

      expect(within(reportRow).getByRole('cell', { name: status }));
      userEvent.click(reportRow);

      const numberOfHistoryCalls = status === 'Locked' ? 0 : 1;
      expect(history.push).toHaveBeenCalledTimes(numberOfHistoryCalls);
    }
  );
});
