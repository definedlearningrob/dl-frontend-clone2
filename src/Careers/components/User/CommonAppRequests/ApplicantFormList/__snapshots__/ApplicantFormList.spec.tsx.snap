// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicantFormList renders correctly 1`] = `
<div>
  <div
    class="shared-table emptyTableWrapper"
  >
    <table
      class="shared-table__table table"
    >
      <thead
        class="shared-table__head shared-table__head-cols-5"
      >
        <tr
          class="shared-table__row shared-table__row-0"
        >
          <th
            class="shared-table__column -header-cell shared-table__column-1 tableHeadCell flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
          >
            Form name
          </th>
          <th
            class="shared-table__column -header-cell shared-table__column-2 tableHeadCell flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
          >
            Type
          </th>
          <th
            class="shared-table__column -header-cell shared-table__column-3 tableHeadCell flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
          >
            Status
          </th>
          <th
            class="shared-table__column -header-cell shared-table__column-4 tableHeadCell flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
          >
            Deadline
          </th>
          <th
            class="shared-table__column -header-cell shared-table__column-5 tableHeadCell flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
          />
        </tr>
      </thead>
      <tbody
        class="shared-table__body shared-table__body-cols-5"
      >
        <tr
          class="shared-table__row shared-table__row-1 group/row tableRow"
          data-testid="list-item"
        >
          <td
            class="shared-table__column shared-table__column-1 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div>
              <div
                class="formName"
              >
                Counselor secondary report
              </div>
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-2 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div
              class="recommenderType"
            >
              Counselor
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-3 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div
              class="font-medium leading-lg flex items-center bg-neutral-200 text-font-secondary rounded-sm p-xxs text-xs gap-xxs"
            >
              <span
                data-state="closed"
              >
                <div
                  class="flex gap-xxs items-center text-font-secondary"
                >
                  <div
                    class="flex"
                  >
                    <div
                      class="flex align-center justify-center grow-0 shrink-0"
                      data-testid="icon"
                    >
                      <div
                        class="h-sm w-sm"
                      />
                    </div>
                  </div>
                  <span>
                    Locked
                  </span>
                </div>
              </span>
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-4 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <span>
              N/A
            </span>
          </td>
          <td
            class="shared-table__column shared-table__column-5 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          />
        </tr>
        <tr
          class="shared-table__row shared-table__row-2 group/row tableRow"
          data-testid="list-item"
        >
          <td
            class="shared-table__column shared-table__column-1 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div>
              <div
                class="formName"
              >
                Counselor optional report
              </div>
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-2 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div
              class="recommenderType"
            >
              Counselor
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-3 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div
              class="font-medium leading-lg flex items-center bg-neutral-200 text-font-secondary rounded-sm p-xxs text-xs gap-xxs"
            >
              <span
                data-state="closed"
              >
                <div
                  class="flex gap-xxs items-center text-font-secondary"
                >
                  <div
                    class="flex"
                  >
                    <div
                      class="flex align-center justify-center grow-0 shrink-0"
                      data-testid="icon"
                    >
                      <div
                        class="h-sm w-sm"
                      />
                    </div>
                  </div>
                  <span>
                    Locked
                  </span>
                </div>
              </span>
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-4 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <span>
              N/A
            </span>
          </td>
          <td
            class="shared-table__column shared-table__column-5 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          />
        </tr>
        <tr
          class="shared-table__row shared-table__row-3 group/row tableRow"
          data-testid="list-item"
        >
          <td
            class="shared-table__column shared-table__column-1 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div>
              <div
                class="kicker sm secondary before:left-0 before:top-0 before:absolute before:h-full before:w-[2px] before:bg-secondary-500 before:inline-block"
              >
                Deadline
              </div>
              <div
                class="formName"
              >
                Counselor optional report 2
              </div>
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-2 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div
              class="recommenderType"
            >
              Counselor
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-3 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div
              class="font-medium leading-lg flex items-center bg-neutral-200 text-font-secondary rounded-sm p-xxs text-xs gap-xxs"
            >
              <span
                data-state="closed"
              >
                <div
                  class="flex gap-xxs items-center text-font-secondary"
                >
                  <div
                    class="flex"
                  >
                    <div
                      class="flex align-center justify-center grow-0 shrink-0"
                      data-testid="icon"
                    >
                      <div
                        class="h-sm w-sm"
                      />
                    </div>
                  </div>
                  <span>
                    Locked
                  </span>
                </div>
              </span>
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-4 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <span
              data-state="closed"
            >
              <span>
                Nov 1, 2023
              </span>
            </span>
          </td>
          <td
            class="shared-table__column shared-table__column-5 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          />
        </tr>
        <tr
          class="shared-table__row shared-table__row-4 group/row tableRow"
          data-testid="list-item"
        >
          <td
            class="shared-table__column shared-table__column-1 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div>
              <div
                class="kicker sm secondary before:left-0 before:top-0 before:absolute before:h-full before:w-[2px] before:bg-secondary-500 before:inline-block"
              >
                Deadline
              </div>
              <div
                class="formName"
              >
                Counselor midyear report
              </div>
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-2 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div
              class="recommenderType"
            >
              Counselor
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-3 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div
              class="font-medium leading-lg flex items-center bg-neutral-200 text-font-secondary rounded-sm p-xxs text-xs gap-xxs"
            >
              <span
                data-state="closed"
              >
                <div
                  class="flex gap-xxs items-center text-font-secondary"
                >
                  <div
                    class="flex"
                  >
                    <div
                      class="flex align-center justify-center grow-0 shrink-0"
                      data-testid="icon"
                    >
                      <div
                        class="h-sm w-sm"
                      />
                    </div>
                  </div>
                  <span>
                    Locked
                  </span>
                </div>
              </span>
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-4 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <span
              data-state="closed"
            >
              <span>
                Nov 2, 2023
              </span>
            </span>
          </td>
          <td
            class="shared-table__column shared-table__column-5 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          />
        </tr>
        <tr
          class="shared-table__row shared-table__row-5 group/row tableRow"
          data-testid="list-item"
        >
          <td
            class="shared-table__column shared-table__column-1 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div>
              <div
                class="formName"
              >
                Counselor final report
              </div>
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-2 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div
              class="recommenderType"
            >
              Counselor
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-3 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div
              class="font-medium leading-lg flex items-center bg-neutral-200 text-font-secondary rounded-sm p-xxs text-xs gap-xxs"
            >
              <span
                data-state="closed"
              >
                <div
                  class="flex gap-xxs items-center text-font-secondary"
                >
                  <div
                    class="flex"
                  >
                    <div
                      class="flex align-center justify-center grow-0 shrink-0"
                      data-testid="icon"
                    >
                      <div
                        class="h-sm w-sm"
                      />
                    </div>
                  </div>
                  <span>
                    Locked
                  </span>
                </div>
              </span>
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-4 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <span>
              N/A
            </span>
          </td>
          <td
            class="shared-table__column shared-table__column-5 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          />
        </tr>
        <tr
          class="shared-table__row shared-table__row-6 group/row tableRow"
          data-testid="list-item"
        >
          <td
            class="shared-table__column shared-table__column-1 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div>
              <div
                class="formName"
              >
                Counselor recommendation
              </div>
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-2 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div
              class="recommenderType"
            >
              Counselor
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-3 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <div
              class="font-medium leading-lg flex items-center bg-neutral-200 text-font-secondary rounded-sm p-xxs text-xs gap-xxs"
            >
              <span
                data-state="closed"
              >
                <div
                  class="flex gap-xxs items-center text-font-secondary"
                >
                  <div
                    class="flex"
                  >
                    <div
                      class="flex align-center justify-center grow-0 shrink-0"
                      data-testid="icon"
                    >
                      <div
                        class="h-sm w-sm"
                      />
                    </div>
                  </div>
                  <span>
                    Locked
                  </span>
                </div>
              </span>
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-4 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          >
            <span>
              N/A
            </span>
          </td>
          <td
            class="shared-table__column shared-table__column-5 flex items-center relative !px-xs xxxl:!px-x first:!pl-base xxxl:first:!pl-md last:!pr-base !py-sm text-xs cursor-pointer"
            data-testid="shared-table-column"
          />
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="emptyFormsMessage"
  >
    There are no forms
  </div>
</div>
`;
